logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
```

Comment:
I suggest pushing the expectation outside of this method. This makes the `threaded_read_client` behave more like its name, though I maybe would call it something like `read_until_data(read_client, time_out)`. wdyt?